sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/enhancedfanout/Config.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.aws2.kinesis.enhancedfanout;
+
+import static org.apache.beam.sdk.io.aws2.kinesis.enhancedfanout.Checkers.checkNotNull;
+
+import java.io.Serializable;
+import java.time.Instant;
+import org.apache.beam.sdk.io.aws2.kinesis.KinesisIO;
+import org.apache.beam.sdk.io.aws2.kinesis.StartingPoint;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Optional;
+import software.amazon.kinesis.common.InitialPositionInStream;
+
+/** This class is immutable. */
+public class Config implements Serializable {
```

Comment:
@psolomin Just a minor note, from a Beam perspective this is rather uncommon. Typically people are familiar with the immutable and serializable Read / Write instances as configuration. I would encourage this as well here and
 keep static constants in the respective place (or use an interface as container to keep them in one place if you prefer)